internal/command/postgres/failover.go

```diff
@@ -128,3 +141,230 @@ func runFailover(ctx context.Context) (err error) {
 	fmt.Fprintf(io.Out, "Failover complete\n")
 	return
 }
+
+func flexFailover(ctx context.Context, machines []*api.Machine, app *api.AppCompact) error {
+	if len(machines) < 3 {
+		return fmt.Errorf("Not enough machines to meet quorum requirements")
+	}
+
+	io := iostreams.FromContext(ctx)
+
+	leader, err := pickLeader(ctx, machines)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(io.Out, "Performing a failover\n")
+
+	primary_region := ""
+	machines_within_primary_region := make([]*api.Machine, 0)
+
+	for _, machine := range machines {
+		if region, ok := machine.Config.Env["PRIMARY_REGION"]; ok {
+			if primary_region == "" || primary_region == region {
+				primary_region = region
+			} else {
+				return fmt.Errorf("Machines don't agree on a primary region. Cannot safely perform a failover until that's fixed")
+			}
+		}
+
+		if machine.Region == primary_region && primary_region != "" {
+			machines_within_primary_region = append(machines_within_primary_region, machine)
+		}
+	}
+	if primary_region == "" {
+		return fmt.Errorf("Could not find primary region for app")
+	}
+
+	newLeader, err := pickNewLeader(ctx, app, machines_within_primary_region)
+	if err != nil {
+		return err
+	}
+
+	// Stop the leader
+	fmt.Println("Stopping current leader... ", leader.ID)
+	machineStopInput := api.StopMachineInput{
+		ID:     leader.ID,
+		Signal: "SIGINT",
+	}
+
+	flapsClient := flaps.FromContext(ctx)
+	err = flapsClient.Stop(ctx, machineStopInput, leader.LeaseNonce)
+	if err != nil {
+		return fmt.Errorf("could not stop pg leader %s: %w", leader.ID, err)
+	}
+
+	fmt.Println("Starting new leader")
+	_, err = flapsClient.Start(ctx, newLeader.ID, newLeader.LeaseNonce)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("Promoting new leader... ", newLeader.ID)
+	err = ssh.SSHConnect(&ssh.SSHParams{
+		Ctx:      ctx,
+		Org:      app.Organization,
+		App:      app.Name,
+		Username: "postgres",
+		Dialer:   agent.DialerFromContext(ctx),
+		Cmd:      "repmgr standby promote --siblings-follow -f /data/repmgr.conf",
+		Stdout:   ioutils.NewWriteCloserWrapper(colorable.NewColorableStdout(), func() error { return nil }),
+		Stderr:   ioutils.NewWriteCloserWrapper(colorable.NewColorableStderr(), func() error { return nil }),
+		Stdin:    nil,
+	}, newLeader.PrivateIP)
+	if err != nil {
+		return err
+
+	}
+
+	// Restart the old leader
+	fmt.Fprintf(io.Out, "Restarting old leader... %s\n", leader.ID)
+	mach, err := flapsClient.Start(ctx, leader.ID, leader.LeaseNonce)
+	if err != nil {
+		return err
```

Comment:
Let's provide some additional context with the error.  This will make it easier to track down any errors that crop up. 

```golang
fmt.Errorf("failed to start machine %d: %w, leader.ID, err)
```